libfwupdplugin/fu-device.c

```diff
@@ -3618,6 +3618,12 @@ fu_device_setup (FuDevice *self, GError **error)
 	/* convert the instance IDs to GUIDs */
 	fu_device_convert_instance_ids (self);
 
+	/* subclassed */
+	if (klass->ready != NULL) {
+		if (!klass->ready (self, error))
+			return FALSE;
+	}
+
```

Comment:
I was kind of thinking it would be a new device phase all together rather than part of setup.  IE 

`fu_device_ready` calls this exact code and has a check for `ready` that could be used potentially.

It's a lot more spaghetti though, so let me know if you don't think that is valuable.

Reply:
I actually did this first, but it's a lot more state machinery we didn't actually need, so I removed it.

Reply:
Really, ->ready is just a ->setup_phase_2

Reply:
OK..